Build/templates/run_tests.yml

```diff
@@ -0,0 +1,104 @@
+steps:
+  # Install different python versions
+  - template: ./install_python.yml
+    parameters:
+      pythonVersion: '3.9.13' # Last version with an installer
+  - template: ./install_python.yml
+    parameters:
+      pythonVersion: '3.10.11' # Last version with an installer
+  - template: ./install_python.yml
+    parameters:
+      pythonVersion: '3.11.9' # Last version with an installer
+  - template: ./install_python.yml
```

Comment:
whats going on here, we install a bunch a different python versions but only use one?

Reply:
We install a bunch of python versions so we can test with them all. The glass setup uses these versions to generate tests for each one.

Reply:
We use the 3.12 version to run the scripts though.

Reply:
should we use a strategy?

```yml
- job: BuildAndTest
  displayName: Build and Test with Python
  strategy:
    matrix:
      Python3_9:
        pythonVersion: '3.9.13' # Last version with an installer
      Python3_10:
        pythonVersion: '3.10.11' # Last version with an installer
      Python3_11:
        pythonVersion: '3.11.9' # Last version with an installer
      Python3_12:
        pythonVersion: '3.12.7'
        outputVar: InstallPython312
      Python3_13:
        pythonVersion: '3.13.0'
  steps:
```

Reply:
or something like

```yml
parameters:
  pythonVersions: ['3.9.13', '3.10.11', '3.11.9', '3.12.7', '3.13.0']

jobs:
- template: matrix_job.yml
  parameters:
    versions: ${{ parameters.pythonVersions }}
matrix_job.yml Template
```
```yml

parameters:
  versions: []

jobs:
- job: BuildAndTest
  displayName: Build and Test with Python
  strategy:
    matrix: ${{ convertToMatrix(parameters.versions) }}
  steps:
    - script: |
        echo "Testing Python version ${{ matrix.pythonVersion }}"
      displayName: Test Python Version
```

Reply:
The test runs all versions at the same time though

Reply:
Otherwise we'd be duplicating all the drop, build, and setup for each job

Reply:
Hmm, maybe I could cache the drop and glass folders though.

Reply:
Then run the tests in parallel for each version